### PR TITLE
Search parameters already wrapped in parentheses no longer get wrapped in quotes

### LINKED ISSUE #114: Search parameters wrapped in parentheses should not be auto-quoted
## Expected Behavior
Don't auto-quote search params wrapped in parentheses.
```python
>>> q = ads.SearchQuery(title='("solar" OR "sun" OR "helio" OR "cme" OR "corona")')
>>> q._query
{'q': ' title:("solar" OR "sun" OR "helio" OR "cme" OR "corona")',
 [...] }
```

## Current Behavior
```python
>>> q = ads.SearchQuery(title='("solar" OR "sun" OR "helio" OR "cme" OR "corona")')
>>> q._query
{'q': ' title:"("solar" OR "sun" OR "helio" OR "cme" OR "corona")"',
 [...] }
```

## Possible Solution
Use `bool(re.match('\s*\(.*\)\s*', v))` to determine whether a parameter value is wrapped in parentheses. Don't wrap in quotes if already wrapped in parentheses.

## Context
Trying to structure my ads queries in a JSON file rather than use a long string.

### PR DESCRIPTION
**Pull request by panasenco**
Fixes #114

### CODE REVIEW

File: ads/search.py [hunk 1]
```diff
@@ -437,8 +438,11 @@ def __init__(self, query_dict=None, q=None, fq=None, fl=DEFAULT_FIELDS,
 
             # Format and add kwarg (key, value) pairs to q
             if kwargs:
-                _ = [u'{}:"{}"'.format(k, v) for k, v in six.iteritems(kwargs)]
-                self._query['q'] = u'{} {}'.format(self._query['q'], ' '.join(_))
+                for field, value in six.iteritems(kwargs):
+                    if not re.match('\s*\(.*\)\s*', value):
+                        # Wrap value in quotes if not already in parentheses
+                        value = u'"{}"'.format(value)
+                    self._query['q'] += u' {}:{}'.format(field, value)
```
Comment (vsudilov) [inline ads/search.py:445]: is the leading space here intentional?

**Review by vsudilov — COMMENTED** [review]
Would be great if you could add a test case showing that this works as expected. Otherwise LGTM

File: ads/search.py (hunk 1 above)
Comment (panasenco) [reply ads/search.py:445]: If you mean the leading space in `u' {}:{}'` then yes - spaces separate query arguments, no?

**Comment by panasenco** [comment]
Sure, that's a reasonable request. I'll try to add some tests sometime this week.

File: ads/search.py (hunk 1 above)
Comment (vsudilov) [reply ads/search.py:445]: Yep, that's what I meant. I'd have to deep-dive again to confirm, so thanks for confirming for me!

**Comment by panasenco** [comment]
I think bumping the Python3 version on TravisCI to a more modern one like 3.8 or 3.9 will fix the import issue it's experiencing.

**Comment by vsudilov** [comment]
> I think bumping the Python3 version on TravisCI to a more modern one like 3.8 or 3.9 will fix the import issue it's experiencing.

Yeah for sure.

**Pull request merged**